common/shamir-secret.c

```diff
@@ -0,0 +1,267 @@
+#include <stdio.h>
+#include </opt/homebrew/Cellar/gmp/6.2.1_1/include/gmp.h>
```

Comment:
Hmm, this needs to be fixed in configure?  It needs to set HAVE_GMP, and in that case add GMP_CFLAGS (and maybe GMP_LDFLAGS if we need -lgmp?).